src/project.js

```diff
@@ -155,25 +167,28 @@ function percentify (value, limit) {
     return (value / limit) * 100;
 }
 
+// implementation of floydWarshall alg for calculating visibility matrix in O(n^3) instead of O(n^4) with successive raising of powers
 function createVisibilityMatrix (result) {
-    var product = result.adjacencyMatrix, sum = result.adjacencyMatrix, changeCost = 0, visibilityMatrix;
 
-    result.adjacencyMatrix.forEach(function () {
-        product = matrix.multiply(product, result.adjacencyMatrix);
-        sum = matrix.add(product, sum);
-    });
+    var changeCost = 0, visibilityMatrix = adjacencyToDistMatrix(result.adjacencyMatrix), matrixLen = visibilityMatrix.length, k, i, j;
```

Comment:
Let's pull the call to `adjacencyToDistMatrix` down to it's own line I reckon. It gets a bit lost in all those other assignments and this line has grown too long anyway.